src/window.cpp

```diff
@@ -0,0 +1,73 @@
+#include "window.h"
+#include "SFML/Window.hpp"
+
+Window::Window(int screenWidth, int screenHeight, int frameRate) 
+{
+    window.create(sf::VideoMode(screenWidth, screenHeight), "Flappy Bird");
+
+    sf::Vector2u windowDimensions = window.getSize();
+    pipes = new Pipes(windowDimensions.x, window.getSize().y, frameRate);
+
+    sf::Vector2f center = window.getView().getCenter();
+    bird = new Bird(center, frameRate);
+
+    this->frameRate = frameRate;
+    this->frameTime = sf::seconds(1.0f / frameRate);
+}
+
+void Window::handleEvents()
+{
+    sf::Event event;
+    while (window.pollEvent(event))
+    {
+        if (event.type == sf::Event::Closed)
+        {
+            window.close();
+            return;
+        }
+
+        if (event.type == sf::Event::KeyPressed && event.key.code == sf::Keyboard::Space)
+        {
+            bird->jump();
+            return;
+        }
+
+    }
+}
+
+bool Window::isOpen()
+{
+    return window.isOpen();
+}
+
+void Window::update()
+{
+    // check if timing is right
+    if(clk.getElapsedTime() < frameTime) return;
+
+    clk.restart();
```

Comment:
This is not a good way to hit a target framerate. You are wasting a lot of CPU time doing nothing polling updates and checking if the clock is ready. If you wanted the simplest way to at least make this better, you could sleep your thread but this is not really ideal. Typically games handle this by rendering as fast as possible and separating the game physics update rate from the render rate. Check out the following blog post for more info here: [Fix Your Timestep](https://gafferongames.com/post/fix_your_timestep/)